packages/node_modules/nav-frontend-skjema/src/radio-panel-gruppe.tsx

```diff
@@ -0,0 +1,116 @@
+    label: string;
+    value: string;
+    id: string;
+    disabled?: boolean;
+}
+
+export interface FeilProps {
+    feilmelding: React.ReactNode | React.ReactChild | React.ReactChildren;
+}
+
+export interface RadioPanelGruppeProps {
+    radios: RadioProps[];
+    name: string;
+    legend: string;
+    onChange: (v: string) => void;
+    checked?: string;
+    feil?: FeilProps;
+}
+
+export interface RadioPanelProps extends RadioProps {
+    checked: boolean;
+    name: string;
+    onChange: () => void;
+}
+
+interface RadioPanelState {
+    hasFocus: boolean;
+}
+
+class RadioPanel extends React.Component<RadioPanelProps, RadioPanelState> {
+    constructor(props: RadioPanelProps) {
+        super(props);
+        this.state = { hasFocus: false };
+    }
+
+    toggleOutline() {
+        this.setState({ hasFocus: !this.state.hasFocus });
+    }
+
+    render() {
+        const { checked, disabled, label, name, id, onChange } = this.props;
+        const { hasFocus } = this.state;
+
+        const cls = classNames('radioPanel', {
+            'radioPanel--checked': checked === true && !disabled,
+            'radioPanel--focused': hasFocus === true && !disabled,
+            'radioPanel--disabled': disabled === true
+        });
+
+        return (
+            <div
+                className={cls}
+                onClick={() => {
+                    if (!disabled) {
+                        onChange();
+                    }
+                }}
+            >
+                <input
+                    className="radioPanel__radio"
+                    type="radio"
+                    name={name}
+                    id={id}
+                    checked={checked}
+                    disabled={disabled}
+                    onFocus={() => this.toggleOutline()}
+                    onBlur={() => this.toggleOutline()}
+                    onChange={() => {}}
+                />
+                <label className="radioPanel__label" htmlFor={id}>{label}</label>
+            </div>
+        );
+    }
+}
+
+class RadioPanelGruppe extends React.Component<RadioPanelGruppeProps> {
+    constructor(props: RadioPanelGruppeProps) {
+        super(props);
+        this.radioValueChanged = this.radioValueChanged.bind(this);
+    }
+
+    radioValueChanged(value) {
+        const { onChange } = this.props;
+        onChange(value);
+    }
+
+    render () {
+        const { radios, name, legend, feil, checked } = this.props;
+        return (
+            <SkjemaGruppe className="radioPanelGruppe" feil={feil}>
+                <Fieldset legend={legend}>
+                    {
+                        radios.map((radio: RadioProps) => (
+                            <RadioPanel
+                                name={name}
+                                key={`${name}-${radio.value}`}
+                                checked={checked === radio.value}
+                                onChange={() => this.radioValueChanged(radio.value)}
```

Comment:
Bare ha `this.radioValueChanged` her og og plukke ut `value` fra `event`?

Reply:
Dette vil vel kanskje løse seg om man bytter strukturen, som nevnt i den andre kommentaren? 
Da skal man ihvertfall ha mulighet til å bruke `onChange` som normalt om jeg ikke tar totalt feil.